iocage_lib/ioc_plugin.py

```diff
@@ -84,6 +84,125 @@ def __init__(self, release=None, jail=None, plugin=None, branch=None,
             # Backwards compat
             self.branch = 'master'
 
+    def clone_repo(self, depth=None):
+        if self.server == "download.freebsd.org":
+            git_server = "https://github.com/freenas/iocage-ix-plugins.git"
+        else:
+            git_server = self.server
+
+        git_working_dir = f"{self.iocroot}/.plugin_index"
+
+        if os.geteuid() == 0:
+            try:
+                self._clone_repo(
+                    self.branch, git_server, git_working_dir,
+                    depth, self.callback
+                )
+            except Exception as err:
+                iocage_lib.ioc_common.logit(
+                    {
+                        "level": "EXCEPTION",
+                        "message": err
+                    },
+                    _callback=self.callback
+                )
+
+    @staticmethod
+    def fetch_plugin_packagesites(package_sites):
+        def download_parse_packagesite(packagesite_url):
+            package_site_data = {}
+            try:
+                response = requests.get(f'{packagesite_url}/All/', timeout=20)
+                response.raise_for_status()
+
+                for pkg in re.findall(r'<a.*>\s*(\S+).txz</a>', response.text):
+
+                    pkg, version = pkg.rsplit('-', 1)
+                    epoch_split = version.rsplit(',', 1)
+                    epoch = epoch_split[1] if len(epoch_split) == 2 else '0'
+                    revision_split = epoch_split[0].rsplit('_', 1)
+                    revision = \
+                        revision_split[1] if len(revision_split) == 2 else '0'
+                    package_site_data[pkg] = {
```

Comment:
How about moving this (parsing `pkg` into dict) to to separate function and creating some unit tests?